### PR TITLE
wip: Core: Toolbar Enhancements

### PR DESCRIPTION
**Pull request by olshansky**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/2890
#### Please check whether the PR fulfills the following requirements

- [ ] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [ ] tests for the changes that have been done
- [ ] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [ ] update `README.md`
- [ ] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [ ] Documentation Examples
- [ ] Stackblitz works for all examples

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/toolbar/toolbar-docs.component.html [hunk 1]
```diff
@@ -33,6 +33,43 @@
 <code-example [exampleFiles]="toolbarOverflowExample"></code-example>
 <separator></separator>
 
+<fd-docs-section-title [id]="'Toolbar-overflow-priority'" [componentName]="'toolbar'">
```
Comment (valorkin) [inline apps/docs/src/app/core/component-docs/toolbar/toolbar-docs.component.html:36]: unnecessary square bindings
```suggestion
<fd-docs-section-title id="Toolbar-overflow-priority" componentName="toolbar">
```

File: libs/core/src/lib/toolbar/toolbar.component.ts [hunk 2]
```diff
@@ -219,19 +279,132 @@ export class ToolbarComponent implements OnInit, AfterViewInit, OnDestroy, After
...
+
+            return lowestGroup;
+        }, itemGroup);
     }
 
     /** @hidden */
     private _removeToolbarItemFromDOM(toolbarItem: ToolbarItemDirective): void {
-        // IE11 workaround element.remove() is not supported
-        toolbarItem.elementRef.nativeElement.parentNode.removeChild(toolbarItem.elementRef.nativeElement);
+        if (toolbarItem.elementRef.nativeElement.parentNode) {
```
Comment (valorkin) [inline libs/core/src/lib/toolbar/toolbar.component.ts:404]: `nativeElement`

File: libs/core/src/lib/toolbar/toolbar.component.ts [hunk 3]
```diff
@@ -219,19 +279,132 @@ export class ToolbarComponent implements OnInit, AfterViewInit, OnDestroy, After
...
+            // IE11 workaround element.remove() is not supported
+            toolbarItem.elementRef.nativeElement.parentNode.removeChild(toolbarItem.elementRef.nativeElement);
```
Comment (valorkin) [inline libs/core/src/lib/toolbar/toolbar.component.ts:406]: https://angular.io/api/core/Renderer2#removechild

File: libs/core/src/lib/utils/consts/overflow-priority-score.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export const OVERFLOW_PRIORITY_SCORE = new Map([
+    ['always', 0],
```
Comment (dimamarksman) [inline libs/core/src/lib/utils/consts/overflow-priority-score.ts:2]: Shouldn't be typed to something like `Map<OverflowPriority, number>`?

**Comment by olshansky** [comment]
https://github.com/SAP/fundamental-ngx/pull/3019 merged

**Pull request closed without merging**